### ISSUE TITLE
Lists and SkyCoord do not scale well

### ISSUE THREAD

**Issue by zoghbi-a** [nasa-fornax/fornax-demo-notebooks#179]
I mentioned this in a comment elsewhere, but I think it should be a separate issue.

Lists and `SkyCoord` don't seem to scale well for a large number of sources (>1e5).

Switching from lists to astropy Tables (PR #176) does part of the job.

Using SkyCoord for every row is slow. This figure compares `SkyCoord` vs using columns of `RA` and `DEC` directly for `get_SDSS_sample` as an example.

![get_sdss_benchmark](https://github.com/fornax-navo/fornax-demo-notebooks/assets/32649422/89c50812-1d4c-4e67-89a2-4b0152424b54)

There is 2 orders of magnitude speed up. 

I suggest dropping the use of `SkyCoord` for individual rows. If there is a need for unit conversion (unlikely), it can be applied to columns and then saved back as columns of numbers  not `SkyCoord` objects.

**Comment by bsipocz** [comment]
> I suggest dropping the use of SkyCoord for individual rows.

You mean dropping iterating on rows? E.g. converting the ra/dec columns into a SkyCoord mixin column may still worth it if it can be directly fed into queries as a vector, doesn't it?

**Comment by zoghbi-a** [comment]
Yes. Iterating over `SkyCoord` objects is the slow part. I am not sure about the  `SkyCoord` columns. I think it should be fine as long its internal operations don't iterate over `SkyCoord` objects.

As far as I can see,`SkyCoord` are needed only in the table join operations to remove duplicates. Constructing a `SkyCoord` column from ra/dec columns there should be enough.

**Comment by bsipocz** [comment]
I think we tried to remove all object in list constructs, but there may be a few accidental leftovers (e.g. column of scalar skycoord as opposed to mixin skycoords). That being said, this problem maybe somewhat related to the issue we discussed last week with @troyraen and @jkrick (and which issue I cannot find somehow to xref here): when to switch over to use a pandas DF vs an astropy Table. The former won't work properly with skycoords or units while the latter maybe more familiar to the user community.

**Comment by troyraen** [comment]
I don't think this is related to pandas dataframes as I don't think we're using using them for any of these types of calls. Sounds to me like the issue comes from iterating over individual objects rather than doing vectorized queries? That would match my experience.

#176 gets rid of the last of the coords lists so `sample_table` will be used for everything after that. However, TESS_Kepler_functions.py and panstarrs.py still iterated over individual objects. This is a known issue and those functions are still being worked on.

**Comment by bsipocz** [comment]
afaik, pandas cannot handle skycoord columns, thus using it in a vectorized way is not really feasible (unless it starts with the extra step of creating Skycoord from the ra/dec coulumns first and then using it in the query. )

**Comment by troyraen** [comment]
We are not using pandas for this.

**Comment by bsipocz** [comment]
But wasn't the plan to switch over? I can't find the issue now, but recall that it was raised that users may like to switch to DF earlier than we do now.

**Comment by troyraen** [comment]
I think this is essentially a duplicate of #165 and #166. Can we close this, just to clean up the issues?

**Comment by zoghbi-a** [comment]
I am fine closing this, but I don't see discussion in those issues, maybe it is useful to keep the discussion that may help in speeding things up.

**Comment by troyraen** [comment]
You're right, there are specific things here that haven't been addressed yet. Thanks for tracking them down.
